examples/reddit.py

```diff
@@ -99,7 +55,8 @@ def train(epoch):
 @torch.no_grad()
 def test():
     model.eval()
-    y_hat = model.inference(data.x, subgraph_loader).argmax(dim=-1)
```

Comment:
@rusty1s  This is because the previous version set all data.x in GPU.
And current `GraphSAGE` class  will set all data.x in CPU, `x_all = loader.data.x.cpu() `
When I use `    y_hat = model.inference(data.x.cpu(), subgraph_loader).argmax(dim=-1)` in the previous version(6.96it/s), it is slower as the newer version (7.31it/s).

Reply:
This should fix the slow inference #7548